tensorflow_addons/layers/python/normalizations.py

```diff
@@ -0,0 +1,363 @@
+# Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# =============================================================================
+
+# Orginal implementation from keras_contrib/layer/normalization
+
+from tensorflow.keras import backend as K
+from tensorflow.keras import constraints
+from tensorflow.keras import initializers
+from tensorflow.keras import regularizers
+from tensorflow.keras.layers import InputSpec
+from tensorflow.keras.layers import Layer
+from tensorflow.python.ops import nn
+
+
+class GroupNormalization(Layer):
+    """Group normalization layer.
+
+    Group Normalization divides the channels into groups and computes
+    within each group the mean and variance for normalization.
+    Group Normalization's computation is independent
+    of batch sizes, and its accuracy is stable in a wide range of batch sizes.
+
+    Relation to Layer Normalization:
+    If the number of groups is set to 1, then this operation becomes identical to
+    Layer Normalization.
+
+    Relation to Instance Normalization:
+    If the number of groups is set to the
+    input dimension (number of groups is equal
+    to number of channels), then this operation becomes
+    identical to Instance Normalization.
+
+    Arguments
+        groups: Integer, the number of groups for Group Normalization.
+            Can be in the range [1, N] where N is the input dimension.
+            The input dimension must be divisible by the number of groups.
+        axis: Integer, the axis that should be normalized
+            (typically the features axis).
+            For instance, after a `Conv2D` layer with
+            `data_format="channels_first"`,
+            set `axis=1` in `BatchNormalization`.
+        epsilon: Small float added to variance to avoid dividing by zero.
+        center: If True, add offset of `beta` to normalized tensor.
+            If False, `beta` is ignored.
+        scale: If True, multiply by `gamma`.
+            If False, `gamma` is not used.
+            When the next layer is linear (also e.g. `nn.relu`),
+            this can be disabled since the scaling
```

Comment:
I think statement like this (`this can be disabled`) is ambiguous, empirical, and not suitable for documentation. Same comments for other layers with this statement.

The true statement is: when the next layer is of certain types (e.g., linear layers), for any trained model with `scale` enabled, there is a model that produces identical inference output with `scale` disabled.

This is far from "this can be disabled". Disabling this in fact often has a significant negative effect on training.

Reply:
> This is far from "this can be disabled". Disabling this in fact often has a significant negative effect on training.

@ppwwyyxx Just a quick question on this quote: 
So disabling the scale vector can lead to an negative effect on training even if the next layer is e.g. a linear layer?

Reply:
Empirically, yes.

Reply:
Oh okay. Thank you for your help. In that case I will remove this section.